models/packages/debian/search.go

```diff
@@ -62,28 +61,40 @@ func SearchLatestPackages(ctx context.Context, opts *PackageSearchOptions) ([]*p
 		})
 	}
 
-	cond = cond.
-		And(builder.Expr("pv2.id IS NULL"))
+	return cond
+}
 
-	joinCond := builder.
-		Expr("package_version.package_id = pv2.package_id AND (package_version.created_unix < pv2.created_unix OR (package_version.created_unix = pv2.created_unix AND package_version.id < pv2.id))").
-		And(builder.Eq{"pv2.is_internal": false})
+// ExistPackages tests if there are packages matching the search options
+func ExistPackages(ctx context.Context, opts *PackageSearchOptions) (bool, error) {
+	return db.GetEngine(ctx).
+		Table("package_file").
+		Join("INNER", "package_version", "package_version.id = package_file.version_id").
+		Join("INNER", "package", "package.id = package_version.package_id").
+		Where(opts.toCond()).
+		Exist(new(packages.PackageFile))
+}
 
-	pfs := make([]*packages.PackageFile, 0, 10)
-	err := db.GetEngine(ctx).
+// SearchPackages gets the packages matching the search options
+func SearchPackages(ctx context.Context, opts *PackageSearchOptions, iter func(*packages.PackageFileDescriptor)) error {
+	return db.GetEngine(ctx).
 		Table("package_file").
 		Select("package_file.*").
 		Join("INNER", "package_version", "package_version.id = package_file.version_id").
-		Join("LEFT", "package_version pv2", joinCond).
 		Join("INNER", "package", "package.id = package_version.package_id").
-		Where(cond).
-		Desc("package_version.created_unix").
-		Find(&pfs)
-	if err != nil {
-		return nil, err
-	}
+		Where(opts.toCond()).
+		Asc("package.lower_name", "package_version.created_unix").
```

Comment:
Is that the correct sorting?
Do we want the sorting
```go
A (yesterday)
A (today)
B (yesterday)
B (today)
```
or do we want
```go
A (today)
B (today)
A (yesterday)
B (yesterday)
```
?

Reply:
In theory we don't need an order here at all because that file is consumed by the apt client which populates its database with the entries.
The sorting is just for readability if a human has a look at the index.